simulation.cpp

```diff
@@ -265,6 +272,15 @@ std::function<double(std::vector<double>)> get_current_env_function(const simula
     return e.get_current_function();
 }
 
+void perform_environment_change(simulation &s)
+{
+  switch_optimal_function(s);
+
+  if(s.get_input().size() > 1){
```

Comment:
As I said above I would also move this if into `switch_optimal_function`

Reply:
I see them as separate processes. I think it makes more sense to have them separated.

Reply:
Mmm, you mean that one step is to switch the function and the other is to actually notify input that it is switched in case we have the "env_func" input? I do not see it that way but I think that is okay, I think your view is valid. But then I would wrap this if statement in a function of its own.